### PR TITLE
Use the same action for toolchain

### PR DESCRIPTION
**Pull request by rakanalh**
# Description
Also remove redundant rustup install commands

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -75,13 +75,9 @@ jobs:
           version: "23.2"
           repo-token: ${{ secrets.GITHUB_TOKEN }}
       - name: Toolchain
-        uses: actions-rs/toolchain@v1
+        uses: dtolnay/rust-toolchain@nightly
```
Comment (kpp) [inline .github/workflows/checks.yml:78]: We need both stable and nightly

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -75,13 +75,9 @@ jobs:
...
         with:
-          toolchain: stable
-          override: true
           components: rustfmt, clippy
-      - name: Install Rust
-        run: rustup show && rustup install nightly && rustup component add rustfmt --toolchain nightly-x86_64-unknown-linux-gnu # Nightly is needed for our configuration of cargo fmt
```
Comment (rakanalh) [inline .github/workflows/checks.yml:84]: @kpp i am basing my command on this.

File: .github/workflows/checks.yml (hunk 1 above)
Comment (rakanalh) [reply .github/workflows/checks.yml:78]: For lints? that's not actually the case. See my other comment

**Comment by kpp** [comment]
Does make test also requires only nightly?

**Comment by rakanalh** [comment]
> Does make test also require only nightly?

I have made it explicit now

**Review by kpp — APPROVED** [review]
LGMT but I think there is no need to add toolchain: stable for a tagged @stable toolchain =)

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**